TOOLS/lua/autocrop.lua

```diff
@@ -235,14 +307,28 @@ end
 
 function on_toggle()
 
+    -- If it's not a video, exit.
+    if not is_playing_video() then
+        mp.msg.info("autocrop is only working for videos.")
+        return
+    end
+
+
+    -- If it is during auto_delay, kill the timer.
+    if timers.auto_delay then
+        timers.auto_delay:kill()
+        timers.auto_delay = nil
+    end
+
+
     -- Cropped => Remove it.
     if remove_filter(labels.crop) then
         return
     end
 
 
     -- Detecting => Leave it.
-    if is_filter_present(labels.cropdetect) then
+    if timers.detect_crop then
```

Comment:
this causes a race condition when executed after the timer has been killed but before the filter is removed in `detect_end`. It could theoretically remove the `cropdetect` right after it is inserted in `detect_crop`